debug/test/browser/debug-compat.test.js

```diff
@@ -0,0 +1,39 @@
+import { createElement, render } from 'preact/compat';
+import { initDebug } from '../../src/debug';
+import { setupScratch, teardown } from '../../../test/_util/helpers';
+
+/** @jsx createElement */
+
+initDebug();
+
+describe('debug with compat', () => {
+	let scratch;
+	let errors = [];
+	let warnings = [];
+
+	beforeEach(() => {
+		errors = [];
+		warnings = [];
+		scratch = setupScratch();
+		sinon.stub(console, 'error').callsFake(e => errors.push(e));
+		sinon.stub(console, 'warn').callsFake(w => warnings.push(w));
+	});
+
+	afterEach(() => {
+		(console.error).restore();
+		console.warn.restore();
+		teardown(scratch);
+	});
+
+	// TODO: This test only passed before because the old test first rendered
+	// a string ref and expected an error. Then it rerendered a vnode with the
+	// same string ref and did not expect an error. The rerender didn't throw
+	// an error because the diff saw the oldVNode and the newVNode as having
+	// the same ref ("a") and so it didn't try to apply it. It had nothing to do
+	// with the "$$typeof" property being present which the old test seems to imply
+	it.skip('should not print an error on string refs', () => {
```

Comment:
Preact core doesn't support string refs and compat doesn't add support for it. No one has complained yet so should we leave this feature out for now and only worry about bringing it back if someone needs it?

Reply:
We deprecated string refs with X. I was the one who removed it :D We got 2 users who noticed so far, but they could easily update by switching to a function callback or the `ref` object. Therefore I think it's safe to remove this test :+1: